README.md

```diff
@@ -1,130 +1,214 @@
 5. Your application's backend server will need to accept the authorization code and exchange it for an access token.
 
-The SDK will help facilitate the OAuth link generation, popup dialog creation, and Smartcar will handle the user authentication and authorization. This SDK will not assist with the backend server code to accept authorization codes or exchanging for access tokens (step 6).
+For a more detailed explanation of our Smartcar-hosted redirect scheme see our [blog post](TODO: LINK HERE) on the topic.
+
+#### Self-Hosted
+
+Here's an example flow should you host the redirect yourself (many of the steps remain similar):
+
+1. User clicks "Connect your car" button (or similar) on your application's website.
+2. User is redirected to an OAuth authentication page, using the `openDialog`, `addClickHandler` or `generateLink` methods. This page requires the user to authenticate with their vehicle credentials.
+3. After entering credentials the user will be presented with a list of their vehicles to authorize and a selection of permissions to allow your app.
+4. Smartcar's services will redirect back to your application's redirect URI. The redirect URI will include a `code` query parameter holding the authorization code if the user approved the authorization. If they denied access the URI will include `error` & `error_description` parameters. If you source `redirect.js` on your redirect page it will trigger execution of your optionally provided `onComplete` method (passed to the `Smartcar` constructor on the client) and close out the redirect page.
+
+    *Note:* in the Smartcar-hosted scheme `redirect.js` handles extracting the query parameters from the redirect URI. In a manual hosting scheme you can choose to do this extraction client side (making use of `redirect.js` if you'd like) or server side. If they approved access you'll need to exchange the `code` parameter for an access token. If they denied access you'll need to handle the error (passed in the `error` argument).
+5. Your application's backend server will need to accept the authorization code and exchange it for an access token.
+
+## Next Steps
+
+This SDK facilitates OAuth link generation, click handler creation and popup dialog creation. This SDK will not assist in exchanging authorization codes for an access token. Should you choose to self-host the redirect this SDK will not assist with extracting the authorization code query parameter from the redirect URI.
```

Comment:
click handler **creation,** and **pop-up** dialog creation.

**If** you choose to self-host the **redirect,** this SDK will not assist **you** with